authorization.py

```diff
@@ -0,0 +1,19 @@
+from fastapi import Header, HTTPException
+
+# Define available roles
+VALID_ROLES = {"admin", "user"}
+
+# Role-based authentication dependency
+def role_required(allowed_roles: list[str]):
```

Comment:
Try to use     from typing import List and List[str] for better compatibility for type check.
Add docstring inside role_required and inner wrapper function.